README.md

```diff
@@ -64,10 +65,9 @@ func main() {
 
     myLogger.Log("Message 2")
 
-    // Configure options, level and app with specific logs
+    // Configure options with specific logs
     newOptions := logger.Options{Level: "warning", Hostname: "gotest", App: "myotherapp", IPAddress: "10.0.1.101", MacAddress:  "C0:FF:EE:C0:FF:EE"}
     errWithOpts := myLogger.LogWithOptions("Message 3", newOptions)
-    errLevelAndApp := myLogger.LogWithLevelAndApp("Message 4", "fatal", "gotest2")
```

Comment:
Why do we want to do away with LogWithLevelAndApp and replace it with LogWithOptions- I can conceive of use cases where the user might want that level of granularity

Reply:
So we already had `LogWithOptions`, which provides the same functionality/granularity - the main reason for removing `LogWithLevelAndApp` was just that it seemed unnecessary when if needed you can do:

```go
logger.LogWithOptions("Message", logger.Options{Level: "debug", App: "foo"})
```

`LogWithLevel` is a common use-case, so it makes sense to keep as a convenience method IMO. For the other potential combinations (hypothetically, `LogWithLevelAndEnv`, or `LogWithLevelAndAppAndEnv`) it doesn't really provide any value for the user since it can be done explicitly with `LogWithOptions`. It also complicates the API of the logger. 

Hopefully that makes sense, definitely open to opinions or use-cases I'm missing. Some of this ties into the changes around Options and the `merge()` method which I'll address below.

Reply:
you'll also notice `LogWithLevel` just calls `LogWithOptions` now too. basically `LogWithOptions` and the reworked handling for message-level options makes everything a user might want to do possible.

Reply:
Makes sense. I think we could add examples like the following to the README to show how LogWithOptions can be used to log env, level, app, etc. 

`logger.LogWithOptions("Message", logger.Options{Level: "debug", App: "foo"})`

Reply:
agreed - I include in LOG-7109 some work to update/refresh the README as a whole as part of adding examples